src/seq_join.rs

```diff
@@ -0,0 +1,361 @@
+use futures::{stream::TryCollect, Future, Stream, TryStreamExt};
+use pin_project::pin_project;
+use std::{
+    collections::VecDeque,
+    future::IntoFuture,
+    iter::Fuse,
+    num::NonZeroUsize,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+/// This helper function might be necessary to convince the compiler that
+/// the return value from [`seq_try_join_all`] implements `Send`.
+/// Use this if you get higher-ranked lifetime errors that mention `std::marker::Send`.
+// pub fn assert_send<'a, F, O>(
+//     fut: F,
+// ) -> F where F: Future<Output = O> + Send + 'a {
+//     fut
+// }
```

Comment:
I should remove this.  It is basic desugaring as far as I can tell, but it doesn't work.